pandas/tests/indexes/numeric/test_indexing.py

```diff
@@ -424,6 +424,17 @@ def test_where_uint64(self):
         result = idx.putmask(~mask, other)
         tm.assert_index_equal(result, expected)
 
+    def test_where_does_not_infer_dtype_from_other(self):
```

Comment:
Good to mention something about string/float inference in this name.

Reply:
OK, next try. How about this name?
```python
def test_where_infers_type_instead_of_trying_to_convert_string_to_float(self):
        # GH 32413
        index = Index([1, np.nan])
        cond = index.notna()
        other = Index(["a", "b"], dtype="string")

        expected = Index([1.0, "b"])
        result = index.where(cond, other)

        tm.assert_index_equal(result, expected)
```
I also adjusted the test, because `other = "a" + Index(range(2)).astype(str)`, had dtype `object` instead of `string`.

Reply:
Yup that works.